app/src/main/java/org/sopt/official/feature/main/MainActivity.kt

```diff
@@ -21,9 +36,161 @@ class MainActivity : AppCompatActivity() {
+                        binding.memberGeneration.generation3.setVisible(generationList.size >= 4)
+                        if (generationList.size >= 4) binding.memberGeneration.generation3.text = generationList[3].toString()
+                        binding.memberGeneration.generation4.setVisible(generationList.size >= 5)
+                        if (generationList.size >= 5) binding.memberGeneration.generation4.text = generationList[4].toString()
+                        binding.memberGeneration.generation5.setVisible(generationList.size >= 6)
+                        if (generationList.size >= 6) binding.memberGeneration.generation5.text = generationList[5].toString()
+                        binding.memberGeneration.generationAddition.setVisible(generationList.size >= 7)
+                        val additionalGeneration = (generationList.size - 5).toString()
+                        if (generationList.size >= 7) binding.memberGeneration.generationAddition.text =
+                            this@MainActivity.getStringExt(R.string.main_additional_generation, additionalGeneration)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun initBlock() {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                launch {
+                    viewModel.blockItem.collect {
+                        val items = it.get()
+                        if (items == null) {
+                            binding.largeBlock.root.setVisible(false)
+                            binding.smallBlock1.root.setVisible(false)
+                            binding.smallBlock2.root.setVisible(false)
+                        } else {
+                            setLargeBlock(items.first)
+                            setSmallBlock(binding.smallBlock1, items.second)
+                            setSmallBlock(binding.smallBlock2, items.third)
+                        }
+                    }
+                }
+                launch {
+                    viewModel.blockList.collect {
+                        smallBlockAdapter?.submitList(it)
+                    }
+                }
+
+            }
+        }
+    }
+
+    private fun initContent() {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                launch {
+                    viewModel.userState.collect { userState ->
+                        val isClickable = userState.get()?.let {
+                            it != UserState.UNAUTHENTICATED
+                        } ?: false
+                        if (isClickable) {
+                            val intent = Intent(applicationContext, SoptampActivity::class.java)
+                            binding.contentSoptamp.root.setOnClickListener {
+                                CoroutineScope(Dispatchers.Main).launch {
```

Comment:
```suggestion
                                lifecycleScope.launch {
```

액티비티 꺼지면 자동으로 취소될 수 있게 설정해주면 좋을 듯!

Reply:
오 저런게 있구만